detox/android/detox/src/full/java/com/wix/detox/DetoxManager.java

```diff
@@ -24,42 +26,47 @@
  */
 class DetoxManager implements WebSocketClient.ActionHandler {
 
-    private final static String LOG_TAG =  "DetoxManager";
-
-    private final static String DETOX_SERVER_ARG_KEY = "detoxServer";
-    private final static String DETOX_SESSION_ID_ARG_KEY = "detoxSessionId";
+    private static final String LOG_TAG =  "DetoxManager";
 
-    private final static Function1<Throwable, String> errorParseFn = new Function1<Throwable, String>() {
+    private static final Function1<Throwable, String> errorParseFn = new Function1<Throwable, String>() {
         @Override
         public String invoke(Throwable t) {
             return Log.getStackTraceString(t);
         }
     };
 
-    private String detoxServerUrl;
-    private String detoxSessionId;
+    private static final ThreadFactory threadFactory = new ThreadFactory() {
+        private int threadCount = 0;
+        @Override
+        public Thread newThread(@Nonnull Runnable r) {
+            String name = "detox" + (threadCount++);
+            return new Thread(r, name);
+        }
+    };
+
+    private static final Integer MAX_CONCURRENT_JOBS = 2;
+    private static final LaunchArgs launchArgs = new LaunchArgs();
 
+    private ExecutorService executor;
     private WebSocketClient wsClient;
-    private Handler handler;
 
-    private TestEngineFacade testEngineFacade = new TestEngineFacade();
-    private Map<String, DetoxActionHandler> actionHandlers = new HashMap<>();
+    private final TestEngineFacade testEngineFacade = new TestEngineFacade();
+    private final Map<String, DetoxActionHandler> actionHandlers = new HashMap<>();
     private ReadyActionHandler readyActionHandler = null;
 
-    private Context reactNativeHostHolder;
+    private final Context reactNativeHostHolder;
+
+    private final String detoxServerUrl;
+    private final String detoxSessionId;
 
     DetoxManager(@NonNull Context context) {
         this.reactNativeHostHolder = context;
 
-        handler = new Handler();
-
-        Bundle arguments = InstrumentationRegistry.getArguments();
-        detoxServerUrl = arguments.getString(DETOX_SERVER_ARG_KEY);
-        detoxSessionId = arguments.getString(DETOX_SESSION_ID_ARG_KEY);
+        detoxServerUrl = launchArgs.getDetoxServerUrl();
+        detoxSessionId = launchArgs.getDetoxSessionId();
 
         if (detoxServerUrl == null || detoxSessionId == null) {
-            Log.i(LOG_TAG, "Missing arguments : detoxServer and/or detoxSession. Detox quits.");
-            stop();
+            Log.i(LOG_TAG, "Missing arguments: detoxServer and/or detoxSession. Detox quits.");
```

Comment:
Wouldn't it make sense to:
1) Log it as an error
2) Move it `void start()` implementation... ?

Reply:
Yes, but all in due time

Reply:
Add a TODO, maybe?